src/npc_attack.cpp

```diff
@@ -306,8 +306,14 @@ npc_attack_rating npc_attack_gun::evaluate_tripoint(
         // if friendly fire is on, we don't care too much, though if an available hit doesn't damage them it would be better.
         attitude_mult = npc_attack_constants::attitude_multiplier.at( Creature::Attitude::NEUTRAL );
     }
-    const int distance_penalty = std::max( distance_to_me - 1,
-                                           ( source.closest_enemy_to_friendly_distance() - 1 ) * 2 );
+    // if there are no npcs, the penalty becomes 0
+    int npc_dist_pen = source.closest_enemy_to_friendly_distance();
+    if( npc_dist_pen == INT_MAX ) {
+        npc_dist_pen = 0;
+    } else {
+        npc_dist_pen = ( npc_dist_pen - 1 ) * 2;
+    }
+    const int distance_penalty = std::max( distance_to_me - 1, npc_dist_pen );
```

Comment:
```suggestion
    // if there are no other NPCs near, the other-NPC penalty is ignored
    int npc_dist_pen = source.closest_enemy_to_friendly_distance();
    if( npc_dist_pen == INT_MAX ) {
        npc_dist_pen = std::abs( distance_to_me - 1 );
    } else {
        npc_dist_pen = ( npc_dist_pen - 1 ) * 2;
    }
    const int distance_penalty = std::min( std::abs( distance_to_me - 1 ), npc_dist_pen );
```
The min is so something being far from somebody else friendly doesn't cancel out being close to the NPC, or vice-versa. The abs is to emphasize that melee weapons may be the best way to go if the enemy is too close.